modules/s3/src/xtdb/s3/checkpoint.clj

```diff
@@ -75,15 +75,20 @@
                                                                ::cp/checkpoint-at cp-at}))})
         cp)))
 
-  (cleanup-checkpoint [this  {:keys [tx cp-at]}]
-    (let [s3-dir (format "checkpoint-%s-%s" (::xt/tx-id tx) (xio/format-rfc3339-date cp-at))
-          files (as-> (s3/list-objects this {:path s3-dir :recursive? true}) $
-                  (keep (fn [[type arg]]
-                          (when (= type :object)
-                            arg)) $)
-                  (vec $)
-                  (conj $ s3-dir))]
-      (s3/delete-objects this files)))
+  (cleanup-checkpoint [this {:keys [tx cp-at]}]
```

Comment:
@jarohen Updated the cleanup-checkpoint implementation on both of the s3 checkpoint stores and added a basic sanity test to ensure that edn metadata file is deleted first and even if the rest fails you cannot list the checkpoint as available :)